src/shelter/use-app-lock.hook.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { act } from 'react-test-renderer';
+import { BehaviorSubject } from 'rxjs';
+
+import { mockCorrectPassword } from '../mocks/react-native-keychain.mock';
+import { mockReactNativeToastMessage } from '../mocks/react-native-toast-message.mock';
+import { mockShelter } from './shelter.mock';
+import { useAppLock } from './use-app-lock.hook';
+
+describe('useAppLock', () => {
+  beforeEach(() => {
+    mockShelter.isLocked$.next(true);
+  });
+
+  describe('initial state', () => {
+    it('should be locked on start by default', () => {
+      const { result } = renderHook(() => useAppLock());
+
+      expect(result.current.isLocked).toEqual(true);
+    });
+
+    it('should be unlocked if shelter is unlocked on start', () => {
+      const { result } = renderHook(() => useAppLock());
+
+      act(() => mockShelter.isLocked$.next(false));
+
+      expect(result.current.isLocked).toEqual(false);
+    });
+  });
+
+  describe('password authentication', () => {
+    beforeEach(() => {
+      mockReactNativeToastMessage.show.mockReset();
+    });
+
+    it('should unlock if a correct password is given', () => {
+      const { result } = renderHook(() => useAppLock());
+
+      act(() => result.current.unlock(mockCorrectPassword));
+
+      expect(result.current.isLocked).toEqual(false);
+    });
+
+    it('should show error toast if an incorrect password is given', () => {
+      mockShelter.unlockApp$.mockImplementationOnce(() => new BehaviorSubject<boolean>(false));
+      const { result } = renderHook(() => useAppLock());
+
+      act(() => result.current.unlock('mockIncorrectPassword'));
+
+      expect(result.current.isLocked).toEqual(true);
+      expect(mockReactNativeToastMessage.show).toBeCalled();
```

Comment:
https://github.com/madfish-solutions/templewallet-mobile/pull/222/files#r689318194
the same, with `mockIncorrectPassword`